Doc/library/sqlite3.rst

```diff
@@ -601,25 +601,16 @@ Connection objects
 
       Example:
 
-      .. testcode::
-
-         def dict_factory(cursor, row):
-             d = {}
-             for idx, col in enumerate(cursor.description):
-                 d[col[0]] = row[idx]
-             return d
-
-         con = sqlite3.connect(":memory:")
-         con.row_factory = dict_factory
-         cur = con.execute("SELECT 1 AS a")
-         print(cur.fetchone()["a"])
-
-         con.close()
-
-      .. testoutput::
-         :hide:
+      .. doctest::
 
-         1
+         >>> def dict_factory(cursor, row):
+         ...     col_names = [col[0] for col in cursor.description]
+         ...     return {key: value for key, value in zip(col_names, row)}
+         >>> con = sqlite3.connect(":memory:")
+         >>> con.row_factory = dict_factory
+         >>> for row in con.execute("SELECT 1 as a, 2 as b"):
+         ...     print(row)
+         {'a': 1, 'b': 2}
```

Comment:
Should the connection be explicitly closed afterward, as in the other examples and in the original one here?

Reply:
For this example, I don't think it matters much. But I'm thinking about turning all the row factory examples into a how-to, and for such a how-to, it makes sense to use a more complete example (create a database, fill it with data, extract data using various row factories, and at last close the connection).